SilverSolver/assignment2/binary_tree.py

```diff
@@ -0,0 +1,61 @@
+class BinaryNode:
+
+    def __init__(self, value, parent=None, child1=None, child2=None):
+        self.value  = value
+        self.parent = parent
+        self.child1 = child1
+        self.child2 = child2
+
+    def get_ancestors(self, prev_ancestors=None, value=True):
+        if not prev_ancestors:
+            prev_ancestors = []
+        if not self.parent:
+            return prev_ancestors
+        else:
+            if value:
+                prev_ancestors.append(self.parent.value)
+            else:
+                prev_ancestors.append(self.parent)
+            return self.parent.get_ancestors(prev_ancestors=prev_ancestors, value=value)
+
+    def get_lowest_common_ancestor(self, another_node, value=True):
+        self_anc = self.get_ancestors(value=False)
+        another_anc = another_node.get_ancestors(value=False)
+        low_c_anc = None
+        while(self_anc and another_anc):
+            if self_anc[-1] == another_anc[-1]:
+                low_c_anc = self_anc[-1]
+                self_anc = self_anc[:-1]
```

Comment:
The bracket notation with the colon creates a copy of the list, the copy can get costly, an integer index would avoid the copy.
There is the zip() builtin iterates on two list at the same time, can you use that?